aten/src/ATen/test/lazy_tensor_test.cpp

```diff
@@ -0,0 +1,45 @@
+#include <gtest/gtest.h>
+
+#include <ATen/ATen.h>
+
+using namespace at;
+
+void LazyFree(void *ptr) {
+  // NOLINTNEXTLINE(cppcoreguidelines-no-malloc)
+  free(ptr);
+}
+
+void* LazyMalloc(ptrdiff_t size) {
+  // NOLINTNEXTLINE(cppcoreguidelines-no-malloc)
+  return malloc(size);
+}
+
+struct LazyAllocator final : public at::Allocator {
+  at::DataPtr allocate(size_t size) const override {
+    auto* ptr = LazyMalloc(size);
+    return {ptr, ptr, &LazyFree, at::DeviceType::XLA};
+  }
+  at::DeleterFnPtr raw_deleter() const override {
+    return &LazyFree;
+  }
+};
+
+void LazyTensorTest(c10::DispatchKey dispatch_key, at::DeviceType device_type) {
+  LazyAllocator allocator;
+  auto tensor_impl = c10::make_intrusive<TensorImpl, UndefinedTensorImpl>(
```

Comment:
how does this work?  allocator doesn't appear to be used.

Reply:
yeah, i saw that this was mostly copy-pasted from an existed XLA test. It's not clear to me what the allocator part is testing though?

Reply:
It used to be needed to create a TensorImpl: https://github.com/pytorch/pytorch/commit/9811a4220d8949fee6915a9b7a9878dc53094eed#diff-52b9f97d02c191bc25b39ecb9c81019c0d4393d13360a17a0e484f5eaf31e087, API changes have made it moot. Should we just delete the test then? We have enough coverage from other places.

Reply:
yea, if the test isn't useful just delete it.  if it's still testing something, you could also just remove the allocator part.  I'm fine either way.

Reply:
Ok, I've removed the allocator for now.